### PR TITLE
fix(api): artifact extension and social history all times

### PR DESCRIPTION
**Pull request by Orta21**
Ref. metriport/metriport#2179

Ticket: #_[ticket-number]_

### Dependencies

- Upstream: none
- Downstream: none

### Description

- Update extensions reference to use Related Artifacts - [context](https://metriport.slack.com/archives/C06CQC7GQG3/p1725053017008029)
- Update Social history in MR to use all dates available - [context](https://metriport.slack.com/archives/C06CQC7GQG3/p1724976777096879)

### Testing

_[Regular PRs:]_

- Local
  - [x] Is validated in Healthlake
- Staging
  - [ ] Make sure output is still valid
- Production
  - [ ] Make sure output is still valid
  - [ ] _testing step 2_

_[Release PRs:]_

Check each PR.

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/core/src/external/aws/lambda-logic/bundle-to-html.ts [hunk 1]
```diff
@@ -1400,29 +1400,8 @@ function createObservationSocialHistorySection(observations: Observation[]) {
       const display = getValidCode(observation.code?.coding)[0]?.display ?? "";
       const value = renderSocialHistoryValue(observation) ?? "";
       const observationDate = formatDateForDisplay(observation.effectiveDateTime);
-      const lastItemInArray = acc[acc.length - 1];
```
Comment (Orta21) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html.ts:1403]: All this wasnt being used so i removed it

File: packages/core/src/external/aws/lambda-logic/bundle-to-html.ts [hunk 2]
```diff
@@ -1432,8 +1411,8 @@ function createObservationSocialHistorySection(observations: Observation[]) {
           LOINC_CODE,
         ]),
         value,
-        firstDate: observationDate,
-        lastDate: observationDate,
+        firstDate: observation.effectivePeriod?.start ? firstDate : observationDate,
+        lastDate: observation.effectivePeriod?.end ? lastDate : observationDate,
```
Comment (leite08) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html.ts:1415]: Its confusing (and error prone as we maintain this code) to check a variable and use another one.

Maybe do this instead?

```ts
  const firstDate = observation.effectivePeriod?.start;
  const lastDate = observation.effectivePeriod?.end;
  ...
  firstDate: firstDate ? formatDateForDisplay(firstDate) : observationDate,
  lastDate: lastDate ? formatDateForDisplay(lastDate) : observationDate,
```

**Review by RamilGaripov — APPROVED** [review]
LGTM

File: packages/utils/src/fhir-deduplication/validate-converted-xmls-healthlake.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+      } catch (error) {
+        console.error(`Error processing ${fileName}`);
+        console.error(error);
+      }
+    },
+    { numberOfParallelExecutions: 1, keepExecutingOnError: true }
+  );
+}
+
+async function validateXml(
```
Comment (jonahkaye) [inline packages/utils/src/fhir-deduplication/validate-converted-xmls-healthlake.ts:121]: nit: should be called validateFhir right?

**Review by RamilGaripov — APPROVED** [review]

File: packages/utils/src/fhir-deduplication/validate-converted-xmls-healthlake.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+  try {
+    await fhirApi.post(`Bundle`, signed.body, {
+      headers: signed.headers,
+    });
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (fhirError: any) {
+    const fhirErrorIssues = fhirError.response?.data.issue;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const severIssues = fhirErrorIssues.filter((issue: any) => issue.severity === "error");
```
Comment (RamilGaripov) [inline packages/utils/src/fhir-deduplication/validate-converted-xmls-healthlake.ts:150]: nit: `fhirErrorIssues?.filter`

File: packages/utils/src/fhir-deduplication/validate-converted-xmls-healthlake.ts [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+    console.log(`Found ${severIssues.length} errors`);
+
+    for (const issue of fhirErrorIssues) {
+      if (issue.severity === "error") {
+        const resource = issue.location[0].split("ofType")[1];
```
Comment (RamilGaripov) [inline packages/utils/src/fhir-deduplication/validate-converted-xmls-healthlake.ts:156]: nit: `issue.location?.[0]?.split`

File: packages/utils/src/fhir-deduplication/validate-converted-xmls-healthlake.ts [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        // const isReferenceError = issue.diagnostics.includes("Reference");
+        // const isContainedError = issue.diagnostics.includes("contained");
+
+        // if (isReferenceError || isContainedError) {
+        //   continue;
+        // }
+
+        errorsFromHealthLake.push({
+          resource: resource,
+          location: issue.location[0],
```
Comment (RamilGaripov) [inline packages/utils/src/fhir-deduplication/validate-converted-xmls-healthlake.ts:168]: nit: `location: issue.location?.[0],`

**Review by RamilGaripov — APPROVED** [review]
left a couple nits

**Pull request merged**